src/Models/clock.jl

```diff
@@ -1,16 +1,32 @@
 import Base: show
 
+using Dates: AbstractTime, Nanosecond
+
 """
     Clock{T<:Number}
 
     Clock{T}(time, iteration)
 
-Keeps track of the current `time` and `iteration` number.
+Keeps track of the current `time` and `iteration` number. The `time::T` can be either a number of a `DateTime` object.
 """
-mutable struct Clock{T<:Number}
+mutable struct Clock{T}
        time :: T
   iteration :: Int
 end
 
+Clock(; time, iteration=0) = Clock(time, iteration)
+
 Base.show(io::IO, c::Clock{FT}) where FT =
     println(io, "Clock{$FT}: time = ", prettytime(c.time), ", iteration = ", c.iteration)
+
+function increment_clock!(clock, Δt)
+    clock.time += Δt
+    clock.iteration += 1
+    return nothing
+end
+

```

Comment:
could also call this “tick!” But looks good

Reply:
Love it!